paddle/fluid/platform/device_context.h

```diff
@@ -750,6 +750,7 @@ class MKLDNNDeviceContext : public CPUDeviceContext {
 
   using ExecMap = std::unordered_map<
       void*, std::vector<std::pair<BlobPtr_t<KeyBlob>, KeyBlob::iterator>>>;
+  using ExecShape = std::unordered_map<std::string, std::shared_ptr<ExecMap>>;
```

Comment:
IMHO those structures deserves some nice documentation. Since every time I look at this code I need to decipher what kind of data are stored here and what are the keys in those maps. Please write detailed explanation. Maybe define some types which will help to make this code more readable. like:
```
using ExecKey = void*;
using BlobMapCacheIterPair = std::pair<BlobPtr_t<KeyBlob>, KeyBlob::iterator>;
using ExecBlobMapsVec = std::vector<BlobMapCacheIterPair>;
using ExecBlobMap = std::unordered_map<ExecKey, ExecBlobMapsVec>;
using ExecShapeMap = std::unordered_map<std::string, std::shared_ptr<ExecBlobMap>>;
```